Run WPAuthenticator tests in dedicated step for CI stability

mokagio opened the pull request:
Internal ref peaMlT-19c-p2

## Description

This PR started out as merely setting the test repetition for the WordPressAuthenticator target unit tests as "retry on failure".

The aim was to mitigate CI flakiness we've seen since merging #1500. It's obviously a band aid because retrying failed tests does not solve the failures, but it buy us time to keep removing unused code (and possibly some of those tests) and reduces the noise in CI.

As you can see in [this test build](https://buildkite.com/automattic/woocommerce-ios/builds/28204#_) that run the unit tests 50 times, introducing test repetition greatly increased the likelihood of the unit tests passing.

![image](https://github.com/user-attachments/assets/20cbbb0f-6e35-4f19-8e82-a12a813e5c01)

As I run different experiments in #15141, I stumbled upon the configuration implemented in this PR:

- Configure both the main suite and the WordPressAuthenticator suite with test repetition mode retry on failure
- Run the WordPressAuthenticator unit tests in a standalone CI step, outside of the main test suite

As you can see in https://buildkite.com/automattic/woocommerce-ios/builds/28230 and https://buildkite.com/automattic/woocommerce-ios/builds/28231 this setup resulted green.

<img width="1707" alt="image" src="https://github.com/user-attachments/assets/4fe842ea-6839-43e5-a16c-549dbec14a0c" />

## Steps to reproduce
N.A.

## Testing information
https://github.com/woocommerce/woocommerce-ios/pull/15141 runs the unit tests 50 times: https://buildkite.com/automattic/woocommerce-ios/builds/28231

## Screenshots
N.A.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. — N.A.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

mokagio committed:
Set test repetition mode for WordPressAuthenticator as "retry on failure"

This aims to mitigate CI flakiness we've seen since merging #1500.

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio committed:
Set test repetition mode to "retry on failure" for the main suite

The WooCommerce `UnitTests.xctestplan` is the one that runs the
WordPressAuthenticator tests, not the one for WordPressAuthenticator
that was edited in bd8e98d696f6b7705850aa2b9b174e67608eee8f.

To enable retry on failure and hopefully address CI flakiness, we need
to modify the test plan that actually runs in CI.

mokagio committed:
Run dedicated WordPressAuthenticator tests in CI

mokagio committed:
Use "Automatically manage signing" in WordPressAuthenticator unit tests

This makes it so that code signing is not required when trying to run
the tests from the terminal with `xcodebuild` or Fastlane `scan`.

mokagio committed:
Disable WordPressAuthenticator tests as part of the main unit tests plan

This is because of the flakiness experienced in CI when running them as
part of the main unit tests test plan.

Instead, we'll run the WordPressAuthenticator unit tests in a dedicated
CI step, which https://github.com/woocommerce/woocommerce-ios/pull/15141
has shown to be far more reliable.

staskus commented on .buildkite/pipeline.yml:
```diff
@@ -44,6 +44,30 @@ steps:
+        prelaunch_simulator:true \
+        device:'iPhone 16'
+    plugins: [$CI_TOOLKIT]
+    parallelism: 50
```
As I understand, this line should be removed to prevent it running 50 times?

AliSoftware replied:
Yep, agreed. Good catch.
```suggestion
```

mokagio replied:
😓 good catch @staskus @AliSoftware I was too rushed in my cherry picking!

mokagio committed:
Remove debug `parallelism: 50` that made it into the branch

Co-authored-by: Olivier Halligon <olivier.halligon@automattic.com>

mokagio committed:
Merge branch 'trunk' into mokagio/enable-retry-on-failure-wpauth

staskus reviewed (approved):
👍 Thanks! The changes makes sense to me, it will increase stability of ongoing PRs
